### PR TITLE
Mobile shaders optimizations

### PR DESCRIPTION
**Pull request by guycalledfrank**
Used in Seemore. Slightly lossy. New API:

- scene.gammaCorrection is now int, possible values:
-- pc.GAMMA_NONE (= 0, same as false in previous version)
-- pc.GAMMA_SRGB (same as true in previous version)
-- pc.GAMMA_SRGBFAST (approximated sRGB curve, used in Seemore)

- material.fastTbn (boolean, false by default) - faster, but not precise TBN calculation in shader
- material.normalizeNormalMap (boolean, true by default) - can be switched off for performance, but generally not recommended

### CODE REVIEW

File: src/graphics/graphics_device.js [hunk 1]
```diff
@@ -293,6 +293,8 @@ pc.extend(pc, function () {
                 this.extCompressedTextureS3TC = gl.getExtension('WEBKIT_WEBGL_compressed_texture_s3tc');
             }
 
+            pc._shaderQuality = pc.SHADERQUALITY_HIGH;
```
Comment (daredevildave) [inline src/graphics/graphics_device.js]: This should be on graphicsDevice not a global

**Comment by willeastcott** [comment]
I feel that these flags are too general and hard to explain. Should we, for now, just expose hints on the phong material?

pc.PhongMaterial#fastGamma = true/false;
pc.PhongMaterial#fastTbn = true/false;

Or a bitfield maybe?

**Comment by guycalledfrank** [comment]
The problem here is that gamma code is selected in scene and used to patch many different shaders (e.g. particles), not only PhongMaterial.

**Comment by willeastcott** [comment]
OK, but do you see my point? Your enum naming is reasonably meaningless. What does it mean for a shader to be medium quality? I'd prefer to think of these as hints or individual flags.

**Comment by guycalledfrank** [comment]
I see the point, not sure yet exactly how we should change it though.

**Comment by guycalledfrank** [comment]
Changed the API

**Pull request merged**